<launch>
    <arg name="config" default="example_scene1" />
    <arg name="fetch_robot_description_parameter_server" default="true" />
    <arg name="urdf_chain" default="robot_description" />
    <arg name="root_link" default="" />
    <arg name="tip_link" default="" />
    <arg name="joint_topic_name" default="/joint_states" />
    <arg name="mp_display" default="true" />
    <arg name="cmp_display" default="true" />
    

    <node name="abstract_robot_test" pkg="constrained_manipulability" type="abstract_robot_test" respawn="false" output="screen" clear_params="true" launch-prefix="">
        <rosparam file="$(find constrained_manipulability)/scene_config/$(arg config).yaml" command="load" />
        <param name="fetch_param_server" type="bool" value="$(arg fetch_robot_description_parameter_server)" />
        <param name="kdl_chain_filename" type="string" value="$(arg urdf_chain)" />
        <param name="root" type="string" value="$(arg root_link)" />
        <param name="tip" type="string" value="$(arg tip_link)" />
        <param name="joint_state_topic_name" type="string" value="$(arg joint_topic_name)" />
        <param name="show_cmp" type="bool" value="$(arg cmp_display)" />
        <param name="show_mp" type="bool" value="$(arg mp_display)" />
    </node>
</launch>
